### PR TITLE
Include environment configuration for LINC Hub

### PR DESCRIPTION
**Pull request by aaronkanzer**
Handling some tech debt here -- consolidates some of the forked repositories in lincbrain GitHub org, also allows environment configurations to live under centralized place in `dandi-hub` thanks to the work of @asmacdo 

@asmacdo perhaps want to give your blessing here? 😄 

Cc @kabilar @satra

### CODE REVIEW

File: envs/linc/jupyterhub-overrides.yaml
Comment (asmacdo) [inline envs/linc/jupyterhub-overrides.yaml:8]:
```suggestion
        - "aaronkanzer"
```

**Review by asmacdo — APPROVED** [review]
Looks good to me-- but this would be a good opportunity to fixup the admin_users list issue we hit last time.

The managed file will need to be regenerated after the fixup, but LGTM

File: envs/linc/jupyterhub-overrides.yaml
Comment (aaronkanzer) [reply envs/linc/jupyterhub-overrides.yaml]: Thanks @asmacdo I bumped these values in my most recent commit.

**Comment by asmacdo** [comment]
Blessing bestowed 🙏

File: envs/linc/jupyterhub-overrides.yaml
Comment (kabilar) [inline envs/linc/jupyterhub-overrides.yaml:97]:
```suggestion
```

File: envs/linc/jupyterhub-overrides.yaml
Comment (kabilar) [inline envs/linc/jupyterhub-overrides.yaml:98]:
```suggestion
            /opt/conda/bin/pip install --upgrade lincbrain-cli;
```

Comment (kabilar) [inline envs/linc/jupyterhub-overrides.yaml:98]:
```suggestion
            /opt/conda/bin/pip install --upgrade dandi;
            /opt/conda/bin/pip install --upgrade lincbrain-cli;
```

Comment (kabilar) [reply envs/linc/jupyterhub-overrides.yaml:98]: I deleted my previous comment and am keeping the upgrade command for the `dandi` package.

**Review by kabilar — CHANGES_REQUESTED** [review]
Thank you @aaronkanzer, @asmacdo.  Its great that we can now manage all three JupyterHubs (DANDI, LINC, BICAN) from this single repo.

I have a couple suggestions above to add the LINC Client and remove items that are for the DANDI users.

File: envs/linc/jupyterhub-overrides.yaml
Comment (aaronkanzer) [reply envs/linc/jupyterhub-overrides.yaml:97]: Great catch here -- removing

**Comment by aaronkanzer** [comment]
> Thank you @aaronkanzer, @asmacdo. Its great that we can now manage all three JupyterHubs (DANDI, LINC, BICAN) from this single repo.
> 
> I have a couple suggestions above to add the LINC Client and remove items that are for the DANDI users.

Thanks @kabilar -- I committed your suggestions -- thanks for the review -- @asmacdo let me know once you are 👍 and I'll merge

**Comment by asmacdo** [comment]
@aaronkanzer I think you want to leave the custom ipython kernel

**Comment by kabilar** [comment]
> Thanks @kabilar -- I committed your suggestions -- thanks for the review

Thanks @aaronkanzer.  Will we need to do anything so that these changes are reflected in the JupyterHub deployment?

**Comment by asmacdo** [comment]
@kabilar im not sure, but the singleuser image has the allen sdk ipython kernel baked in

https://github.com/dandi/dandi-hub/blob/main/images/Dockerfile#L47-L50

We may need to have a separate image?

**Comment by asmacdo** [comment]
@kabilar @aaronkanzer I think we should consider leaving the allen ipython kernel update in until we produce new images since the kernel is baked into the original image. I haven't played with those however, so it may not be necessary.

**Comment by kabilar** [comment]
Sounds good

**Comment by aaronkanzer** [comment]
@asmacdo @kabilar mind doing one last mini-review -- kernel is back in (didnt include the datalad call for dandi) -- if you are happy, I'll re-deploy the LINC Hub

File: envs/.DS_Store
Comment (kabilar) [inline envs/.DS_Store:1]: Please remove this file.

File: envs/linc/jupyterhub-overrides.yaml
Comment (kabilar) [inline envs/linc/jupyterhub-overrides.yaml:94]:
```suggestion
```

File: envs/linc/managed-jupyterhub.yaml
Comment (kabilar) [inline envs/linc/managed-jupyterhub.yaml:194]: I am sure you are already planning to update this script, but heads up that this managed script needs to be updated.

**Comment by kabilar** [comment]
Thank you, @aaronkanzer.  A few minor suggestions above.

File: envs/linc/managed-jupyterhub.yaml
Comment (aaronkanzer) [reply envs/linc/managed-jupyterhub.yaml:194]: Sorry Kabi, could you clarify what you mean by "this managed script needs to be updated" -- I assume you mean that I'll need to redeploy?

File: envs/linc/jupyterhub-overrides.yaml
Comment (aaronkanzer) [reply envs/linc/jupyterhub-overrides.yaml]: Updated!

File: envs/.DS_Store
Comment (kabilar) [reply envs/.DS_Store:1]: Thanks

Comment (kabilar) [inline envs/.DS_Store:1]: Hi @aaronkanzer, maybe something hasn't refreshed on my side, but this file still seems to exist.

File: envs/linc/managed-jupyterhub.yaml
Comment (kabilar) [reply envs/linc/managed-jupyterhub.yaml:194]: Hi Aaron, it looks like the `jupyter-overrides.yaml` and `managed-jupyterhub.yaml` are out of sync.  If I am not mistaken, once you run [install.sh](https://github.com/dandi/dandi-hub/blob/main/install.sh) for the deployment, this `managed-jupyterhub.yaml` file will be updated and then you will need to push those changes to this pull request.

Comment (aaronkanzer) [reply envs/linc/managed-jupyterhub.yaml:194]: I think the only change is the tf state that is upload to S3 if I'm not mistaken, but @asmacdo would know with certainty

File: envs/.DS_Store
Comment (aaronkanzer) [reply envs/.DS_Store:1]: hmm not sure why it didn't get removed -- should be removed now

File: envs/linc/managed-jupyterhub.yaml
Comment (asmacdo) [reply envs/linc/managed-jupyterhub.yaml:194]: The first part of the install script does update the managed jupyterhub file from the overrides. When making a PR like this I usually just comment out the deployment part and execute to get them aligned.

We should make that a cleaner process, (would be nice to add a check to ci also) but for now you do need to run that part of the script.

Comment (kabilar) [reply envs/linc/managed-jupyterhub.yaml:194]: Thanks for clarifying, Austin.  Thats an easy fix.

Comment (aaronkanzer) [reply envs/linc/managed-jupyterhub.yaml:194]: @asmacdo let me know if you might have 5 minutes to zoom -- I _think_ I understand you correctly, but I want to make sure I make the proper steps when re-deploying

**Comment by aaronkanzer** [comment]
@kabilar -- @asmacdo and I successfully deployed these changes this AM -- I'll default to one of the `dandi-hub` repo owners for merging this code.

File: envs/linc/managed-jupyterhub.yaml
Comment (aaronkanzer) [reply envs/linc/managed-jupyterhub.yaml]: Resolving this thread, as @asmacdo and I deployed successfully.

**Pull request merged**
